metadata-ingestion/src/datahub/ingestion/source/affirm_artifact.py

```diff
@@ -0,0 +1,143 @@
+import sys
+import os
+import logging
+from dataclasses import dataclass, field
+from typing import Iterable, Sequence
+from ruamel.yaml import YAML
+
+import datahub.emitter.mce_builder as builder
+from datahub.configuration.common import ConfigModel
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.schema_classes import (
+    DatasetPropertiesClass,
+    OwnerClass,
+    OwnershipClass,
+    OwnershipTypeClass,
+)
+
+
+yaml = YAML(typ='rt')
+
+
+@dataclass
+class AffirmArtifact:
+    name: str
+    description: str
+    owner: str
+    privacy_entrypoint: str
+    retention_days: int
+    processing_purposes: Sequence[str]
+
+    def __post_init__(self):
+        if self.privacy_entrypoint is None:
+            self.privacy_entrypoint = ''
+        if self.retention_days is None:
+            self.retention_days = -1
+        if self.processing_purposes is None:
+            self.processing_purposes = []
+
+
+class AffirmArtifactSourceConfig(ConfigModel):
+    directory: str
```

Comment:
Can you add a TODO here to change this to a Git reference instead? I am thinking as we move to getting this automated, we will want to clone the metadata repo during ingestion and extract artifacts from there.

Reply:
done!